src/win/detect_wakeup.c

```diff
@@ -0,0 +1,65 @@
+#include "uv.h"
+#include "internal.h"
+
+static int uv__register_system_resume_callback();
+
+void uv_init_detect_system_wakeup() {
+  /* Try registering system power event callback. This is the cleanest
+   * method, but it will only work on Win8 and above.
+   */
+  uv__register_system_resume_callback();
+}
+
+#define DEVICE_NOTIFY_CALLBACK 2
```

Comment:
can you move all these declarations to winapi.c ? IIRC that's where we keep them all. Also, does this compile on Windows XP with VS 2008? That's the baseline for libuv v1.x. Please don't judge me :-P